catroid/src/main/java/org/catrobat/catroid/merge/MergeTask.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.merge;
+
+import android.content.Context;
+
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.io.XstreamSerializer;
+
+public class MergeTask {
```

Comment:
If a class gets named task i think about the class doing something for me, not holding references of any object.

It could be built up like functions which get first and second project and only return the mergedProject. As far as i understand the "task" doesnt need to know/hold anything about Projects in this persistent way

Reply:
I name it Merge Task because it will take care of anything merge related.  I decided to make it a class so it would be easily extended in the future.